### PR TITLE
add the general query NOT LIKE operator

### PR DESCRIPTION
**Pull request by d-w-moore**
A new class, irods.column.NotLike, the logical negation of irods.column.Like, is introduced. It implements "NOT LIKE" in the GenQuery syntax.

### CODE REVIEW

**Comment by d-w-moore** [comment]
Ready for review.

File: irods/test/query_test.py [hunk 1]
```diff
@@ -541,6 +541,13 @@ def test_rules_query__267(self):
             Rule(self.sess).remove_by_id( results[0][RuleExec.id] )
 
 
+    def test_not_like_operator__issue_443(self):
+        bump = lambda char_: chr(1 + ord(char_))
+        name_search_pattern = '%' + bump(self.obj_name[-1])
+        query = self.sess.query(DataObject).filter( NotLike(DataObject.name, name_search_pattern),
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/test/query_test.py:547]: the query call creates a generator, a sort of iterator that can be either lazily evaluated (paging through results) or all at once (as in conversion to a `list`)

File: irods/test/query_test.py [hunk 2]
```diff
@@ -541,6 +541,13 @@ def test_rules_query__267(self):
             Rule(self.sess).remove_by_id( results[0][RuleExec.id] )
 
 
+    def test_not_like_operator__issue_443(self):
+        bump = lambda char_: chr(1 + ord(char_))
+        name_search_pattern = '%' + bump(self.obj_name[-1])
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/test/query_test.py:546]: yeah, I think so.  Then we'd create no objects, and make sure that the len(list(...results...)) is 0 length, right?

[the remark this answers is not in the document]
Comment (korydraughn) [reply irods/test/query_test.py:546]: Not quite. I was looking at the use of `bump` and thinking couldn't the same result be reached if we searched for a completely bogus value via `NotLike`.

Everything else would stay the same. I also happy if things can be simplified further by removing more code :-)

**Comment by alanking** [comment]
Please resolve open remaining conversations. I think this is pretty much good to go

**Comment by d-w-moore** [comment]
resolved, and squashed

**Review by korydraughn — APPROVED** [review]
Please add the pound when ready.

**Comment by d-w-moore** [comment]
Pound added

**Review by alanking — APPROVED** [review]

**Pull request merged**